### ISSUE TITLE
`is_inside_bazel_cache` don't work if user changed bazel output_base

### ISSUE THREAD

**Issue by chestnutprog** [lalten/rules_appimage#152]
I set ~/.cache as a symlink to other disk due to `/home` is too full. So bazel's output_base followed it and is_inside_bazel_cache don't work properly. Resulting the appimage is full of symlinks. 
The output of `bazel info ouput_base` looks like `/var/cache/bazel/_bazel_xxx/xxxxxxxxxx` on my laptop.

**Comment by lalten** [comment]
Ah, very interesting find! Thanks for reporting. As a workaround, you can stick with v1.7.x for now which I think should not have that problem.
I'll see if I can make some time soon to fix this, but feel free to send a PR in the meantime!

**Comment by lalten** [comment]
I can not repro this on Bazel 7, are you on Bazel 6 still?

Just changing the output_base is no issue. This works:
```
❯ bazel --output_base=~/.cache/bazel2 test --cache_test_results=no //...
```

Even when it's a symlink the tests are green.
```
❯ cd ~/.cache
❯ ln -s bazel2 bazel2_symlink
❯ cd -
❯ bazel --output_base=~/.cache/bazel2_symlink test --cache_test_results=no //...
```
Also with `--noincompatible_strict_action_env` the tests are still green:
```
❯ bazel --output_base=~/.cache/bazel2_symlink test --cache_test_results=no //... --noincompatible_strict_action_env
```

I was only able to repro on Bazel 6:
```
❯ USE_BAZEL_VERSION=6.x bazel --output_base=~/.cache/bazel2_symlink test --cache_test_results=no //...
(...)
==================== Test output for //tests:appimage_test_py:
Traceback (most recent call last):
  File "/tmp/appimage_extracted_0bde69e0e1168114c1b35195bd5bea5b/tests/test_py.runfiles/rules_appimage/tests/test.py", line 141, in <module>
    test_symlinks()
  File "/tmp/appimage_extracted_0bde69e0e1168114c1b35195bd5bea5b/tests/test_py.runfiles/rules_appimage/tests/test.py", line 83, in test_symlinks
    assert link.is_symlink()
AssertionError
================================================================================
```

So you could mitigate the issue by upgrading to Bazel 7.

**Comment by lalten** [comment]
@chestnutprog can you please check if https://github.com/lalten/rules_appimage/pull/155 solves your issue?

**Comment by chestnutprog** [comment]
Hi @lalten , Thanks a lot for your fix PR! I tried it on my laptop, and it works well.
However, I found another issue that symbol links created by https://github.com/jvolkman/rules_pycross still wrongly packed in the appimage. The problem is pycross' depsets only include a single dictionary, so https://github.com/lalten/rules_appimage/blob/main/appimage/private/tool/mkappimage.py#L90 will directly add whole dictionary full of symlinks into new_manifest_data.
I think the best way to solve it may be use `Args.add_all()` to let bazel expand the dictionary, but it is a little hard to me and it make lot of changes. Another way is creating another loop in mkappimage.py to iter on every dictionary.
If you think it is ok to iter on every dictionary, I'd like to draft a PR to solve it~
